.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Pull Request Prerequisite Checks
+
+on:
+  pull_request:
+    branches:
+      - main
+  
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: 0
+
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v2
+        with:
+          main-branch-name: ${{ github.base_ref }}
+
+      - name: Setup Node with version ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: curl -s -u"${{ secrets.MAVEN_USERNAME }}:${{ secrets.MAVEN_PASSWORD }}" https://repo.backbase.com/api/npm/npm-backbase/auth/backbase > .npmrc
```

Comment:
Also, we shouldn’t override the original `.npmrc` file. We can create a custom `.npmrc` (e.g. `.npmrc-user`) file and specify it as a user config file:

```
curl -s -u"${{ secrets.MAVEN_USERNAME }}:${{ secrets.MAVEN_PASSWORD }}" https://artifacts.backbase.com/api/npm/auth | sed "s/^/\/\/artifacts.backbase.com\/api\/npm\/npm-backbase\/:/" >> .npmrc-user
curl -s -u"${{ secrets.MAVEN_USERNAME }}:${{ secrets.MAVEN_PASSWORD }}" https://repo.backbase.com/api/npm/auth | sed "s/^/\/\/repo.backbase.com\/api\/npm\/npm-backbase\/:/" >> .npmrc-user
export NPM_CONFIG_USERCONFIG=.npmrc-user
```

In this way, we keep project's npmrc and add tokens for repo and artifacts registries.

Reply:
Hi Kirill, But what is the issue in creating .npmrc in a container instance? Once the job completes the container will be deleted. As this new created file is not pushed to the repo, I feel this should not be an issue.

FYI: I am not in favour of using curl at all. This seems like a hack. setup-node action already has this functionality where it create a npmrc file for the private registry but needs a auth-token. In our case for repo we don't have such thing.
https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#example-workflow

Reply:
I can add the auth token to the secrets, no problem
We need one for repo/artifactory or the public npmjs.org?

Reply:
@santam85 only for `repo`. We can't download from `artifacts` without VPN.

Reply:
@manisha-backbase we have to keep an original `.npmrc` because it is a part of the source code. We can go with your solution but by appending to the existing `.npmrc` (with `>>` instead `>`)

Reply:
@kirillsud  I have made the changes

Reply:
Token requested with https://backbase.atlassian.net/servicedesk/customer/portal/19/SUPPORT-38709

Reply:
We can change the implementation later on. Once, we get some information to use the auth token for the repo.
I will resolve this conversation for now.